src/language-css/printer-postcss.js

```diff
@@ -507,60 +507,62 @@ function genericPrint(path, options, print) {
-            (iNextNextNode && isWordNode(iNextNextNode)) ||
-            iNode.type === "value-func" ||
-            isWordNode(iNode);
-          const hasSpaceAfterOperator =
-            iNextNode.type === "value-func" ||
-            isWordNode(iNextNode) ||
-            (iPrevNode && iPrevNode.type === "value-func") ||
-            (iPrevNode && isWordNode(iPrevNode));
-          const insideCalcFunction = insideValueFunctionNode(path, "calc");
-
-          if (
-            // Multiplication
-            !isMultiplication &&
-            !isNextMultiplication &&
-            // Division
-            !(
-              isNextDivision &&
-              (hasSpaceBeforeOperator || insideCalcFunction)
-            ) &&
-            !(isDivision && (hasSpaceAfterOperator || insideCalcFunction)) &&
-            // Addition
-            !(isNextAddition && hasSpaceBeforeOperator) &&
-            !(isAddition && hasSpaceAfterOperator)
-          ) {
-            if (
-              hasEmptyRawBefore(iNextNode) ||
-              (isMathOperator &&
-                (isParenGroupNode(iNextNode) ||
-                  isWordNode(iNextNode) ||
-                  iNextNode.type === "value-number" ||
-                  isMathOperatorNode(iNextNode)) &&
-                (!iPrevNode || (iPrevNode && isMathOperatorNode(iPrevNode))))
-            ) {
-              continue;
-            }
-          }
+        // Print spaces before and after math operators beside SCSS interpolation as is
+        // (i.e. `#{$var}+5`, `#{$var} +5`, `#{$var}+ 5`, `#{$var} + 5`)
+        // (i.e. `5+#{$var}`, `5 +#{$var}`, `5+ #{$var}`, `5 + #{$var}`)
+        if (
+          ((isMathOperator && isHashNode(iNextNode)) ||
+            (isNextMathOperator && isRightCurlyBraceNode(iNode))) &&
+          hasEmptyRawBefore(iNextNode)
+        ) {
+          continue;
+        }
+
+        // Print spaces before and after addition and subtraction math operators as is in `calc` function
+        // due to the fact that it is not valid syntax
+        // (i.e. `calc(1px+1px)`, `calc(1px+ 1px)`, `calc(1px +1px)`, `calc(1px + 1px)`)
+        if (
+          insideValueFunctionNode(path, "calc") &&
+          (isAdditionNode(iNode) ||
+            isAdditionNode(iNextNode) ||
+            isSubtractionNode(iNode) ||
+            isSubtractionNode(iNextNode)) &&
+          hasEmptyRawBefore(iNextNode)
+        ) {
+          continue;
+        }
+
+        const requireSpaceBeforeOperator =
+          (iNextNextNode && iNextNextNode.type === "value-func") ||
+          (iNextNextNode && isWordNode(iNextNextNode)) ||
+          iNode.type === "value-func" ||
+          isWordNode(iNode);
+        const requireSpaceAfterOperator =
+          iNextNode.type === "value-func" ||
+          isWordNode(iNextNode) ||
+          (iPrevNode && iPrevNode.type === "value-func") ||
+          (iPrevNode && isWordNode(iPrevNode));
+
+        // Formatting `/`, `+`, `-` sign
```

Comment:
Isn't this `if` about _not_ formatting things? Could you add helpful examples like in the above two `if`s?

Reply:
@lydell i will do second refactor and add message in second PR :+1: